### PR TITLE
Adds docker example

### PR DESCRIPTION
**Pull request by rolandpoulter**
Uses vagrant and boot2docker to allow docker-compose to bring up RackHD with ELK.

Dockerfiles:
~~https://github.com/RackHD/on-core/pull/72~~
~~https://github.com/RackHD/on-dhcp-proxy/pull/23~~
~~https://github.com/RackHD/on-http/pull/110~~
~~https://github.com/RackHD/on-syslog/pull/16~~
~~https://github.com/RackHD/on-tftp/pull/28~~
~~https://github.com/RackHD/on-taskgraph/pull/53~~
~~https://github.com/RackHD/on-statsd/pull/9~~

### CODE REVIEW

**Comment by rolandpoulter** [comment]
@mattgam @heckj @benbp @DavidjohnBlodgett

**Comment by jlongever** [comment]
test this please

**Comment by rolandpoulter** [comment]
* Updated Dockerfiles to build more effeciently.
* Configure isc-dhcp-server from RackHD/docker.
* Added Kibana and setup Logstash to listen to RabbitMQ on.logging.
* Added missing common files to on-http.

**Comment by rolandpoulter** [comment]
* Added convenient script that will checkout my docker branches of RackHD submodules in order to make it easier to run this before it is merged.
* Updated README with basic instructions.

**Comment by rolandpoulter** [comment]
Waiting on automated build to test docker hub example...

**Comment by rolandpoulter** [comment]
Added docker images to docker hub here: https://hub.docker.com/u/roland/

Used my account for now. This will allow people to run this docker example without building all the RackHD docker images.

**Comment by rolandpoulter** [comment]
I've added `on-statsd` which runs Etsy/statsd with a elasticsearch backend.
https://github.com/RackHD/on-statsd/pull/9

**Comment by jlongever** [comment]
test this please

File: docker/build_images.bash [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+set -x
+
+SCRIPT_DIR=$(cd $(dirname $0) && pwd)
+echo "Rebuilding all docker images."
+
+REPOS="on-core on-dhcp-proxy on-http on-syslog on-tftp on-taskgraph"
+for repo in ${REPOS}; do
+    if [ -d "${SCRIPT_DIR}/../${repo}" ]; then
+        pushd "${SCRIPT_DIR}/../${repo}"
+          docker build -t roland/$repo --no-cache=false .
```
Comment (benbp) [inline docker/build_images.bash]: Should do upstream instead of your fork yeah?

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  privileged: true
+  image: rabbitmq:management
+  net: "host"
+
+# RackHD
+
+on-dhcp-proxy: # 68, 68/udp, 4011
+  privileged: true
+  env_file: docker.env
+  image: roland/on-dhcp-proxy
```
Comment (benbp) [inline docker/docker-compose.yml]: fork reference, and below

**Comment by benbp** [comment]
:+1:

**Comment by heckj** [comment]
:+1:

**Comment by rolandpoulter** [comment]
Please wait to merge this until all the docker files have been merged.

**Comment by rolandpoulter** [comment]
* Updated dockerfiles to use Alpine Linux: https://github.com/RackHD/on-core/pull/79
* Fixed issues from latest API changes. Now mounting `/opt/monorail` volume in all RackHD containers.
* Added `on-statsd` to scripts.

**Comment by rolandpoulter** [comment]
New docker images are tiny compared to before. Previously `rackhd/on-http` was about 1.2GB
```
REPOSITORY             TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
rackhd/on-taskgraph    latest              cb0050db791c        29 minutes ago      209.8 MB
rackhd/on-tftp         latest              b27a334fb1e2        29 minutes ago      182.1 MB
rackhd/on-syslog       latest              83d55b09d7b8        29 minutes ago      179.6 MB
rackhd/on-statsd       latest              bf0819e29ecc        29 minutes ago      195.3 MB
rackhd/on-http         latest              b01794c9693b        30 minutes ago      515.4 MB
rackhd/on-dhcp-proxy   latest              d9dd3401d685        31 minutes ago      183 MB
rackhd/on-core         latest              adfc59f72b75        31 minutes ago      142 MB
mongo                  latest              b11eedbc330f        8 days ago          317.4 MB
rabbitmq               management          fce3161a1999        8 days ago          305.1 MB
elasticsearch          latest              5f950105f780        10 days ago         345.7 MB
logstash               latest              72e3ea5d6e55        11 days ago         447.5 MB
kibana                 latest              50e2a548d2d9        11 days ago         259.8 MB
mhart/alpine-node      4                   ea3184dbb53a        2 weeks ago         36.41 MB
```

**Comment by rolandpoulter** [comment]
I also added a nice landing page `/opt/monorail/http/index.html`
image

**Comment by jlongever** [comment]
+1

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# RackHD on Docker instructions
+
+Copyright 2016, EMC, Inc.
+
+Install Docker Toolbox on your system. See https://www.docker.com/products/docker-toolbox
+
+This has only been tested on Mac OS X.
```
Comment (benbp) [inline docker/README.md]: When you get the chance it would be nice to test this on linux :)

File: docker/logstash/config/logstash.conf [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2016, EMC, Inc.
+
+input {
+	rabbitmq {
```
Comment (benbp) [inline docker/logstash/config/logstash.conf]: Curiosity, does logstash require tabs in the conf file?

**Comment by benbp** [comment]
:+1: 

@rolandpoulter do you need this to wait on the current outstanding docker PRs or can I merge?

File: docker/logstash/config/logstash.conf (hunk 4 above)
Comment (rolandpoulter) [inline docker/logstash/config/logstash.conf]: I'll try switching to spaces

File: docker/README.md (hunk 3 above)
Comment (rolandpoulter) [inline docker/README.md]: I should update that. It will work on linux but the RackHD repo must be at the root. I'll change the readme.

**Comment by rolandpoulter** [comment]
@benbp I think this is safe to merge once I make those changes. Should still work with the old images. The alpine linux docker PRs should be merged more or less at the same time. Or the docker build will fail, which might be okay.

File: docker/README.md (hunk 3 above)
Comment (benbp) [inline docker/README.md]: Ok, I'll wait to merge until that @rolandpoulter, but your other stuff is merged.

**Comment by rolandpoulter** [comment]
@benbp fixed whitespace, and updated README for running on a Linux host.

**Comment by jlongever** [comment]
@rolandpoulter @benbp is this good to go?

**Pull request merged**